x-pack/elastic-agent/pkg/agent/application/enroll_cmd.go

```diff
@@ -557,3 +553,31 @@ func getAppFromStatus(status *client.AgentStatus, name string) *client.Applicati
 	}
 	return nil
 }
+
+func safelyStoreAgentInfo(s saver, reader io.Reader) error {
+	err := storeAgentInfo(s, reader)
+	signal := make(chan struct{})
+	backExp := backoff.NewExpBackoff(signal, 200*time.Millisecond, 15*time.Second)
+
+	for err != nil {
+		backExp.Wait()
+		err = storeAgentInfo(s, reader)
```

Comment:
What if this always fails? How does this not loop forever?

Reply:
yeah copy paste i rewrote that part